### PR TITLE
Eth1 mode RPC proxy tests

### PR DESCRIPTION
**Pull request by jframe**
## PR Description
This adds integration and unit tests to the Eth1 RPC proxy feature being ported from EthSigner to Web3Signer.

Code is copied from EthSigner and modified to work with Web3Signer.

Note for the integration tests I had to bring the rest of the RPC framework from EthSigner for the integration tests to run.

## Fixed Issue(s)

Part of #763

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/DefaultTestBase.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2019 ConsenSys AG.
```
Comment (siladu) [inline core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/DefaultTestBase.java:2]: Does this matter?

File: core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+
+    final File keyFile = createKeyFile();
+    final File passwordFile = createFile("password");
+    credentials = WalletUtils.loadCredentials("password", keyFile);
+
+    final Eth1AddressSignerProvider transactionSignerProvider =
+        new Eth1AddressSignerProvider(new SingleSignerProvider(signer(keyFile, passwordFile)));
+
+    // Force TransactionDeserialisation to fail
+    final ObjectMapper jsonObjectMapper = new ObjectMapper();
```
Comment (siladu) [inline core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java:137]: why do we need to force this to fail?

Comment (siladu) [reply core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java:137]: In fact, does the jsonObjectMapper even get used anywhere?

File: core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+    setupWeb3Signer(chainId, "");
+  }
+
+  static void setupWeb3Signer(final long chainId, final String downstreamHttpRequestPath)
+      throws Exception {
+    setupWeb3Signer(chainId, downstreamHttpRequestPath, List.of("sample.com"));
+  }
+
+  static void setupWeb3Signer(
+      final long chainId,
```
Comment (siladu) [inline core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java:123]: Not used?

File: core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+      final String downstreamHttpRequestPath,
+      final List<String> allowedCorsOrigin)
+      throws Exception {
+    clientAndServer = startClientAndServer();
+
+    final File keyFile = createKeyFile();
+    final File passwordFile = createFile("password");
+    credentials = WalletUtils.loadCredentials("password", keyFile);
```
Comment (siladu) [inline core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java:131]: Not used?

File: core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+    waitForNonEmptyFileToExist(portsFile);
+    final int web3signerPort = httpJsonRpcPort(portsFile);
+    RestAssured.port = web3signerPort;
+
+    LOG.info(
+        "Started web3signer on port {}, eth stub node on port {}",
+        web3signerPort,
+        clientAndServer.getLocalPort());
+
+    unlockedAccount =
```
Comment (siladu) [inline core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java:162]: Not used?

File: core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java [hunk 6]
```diff
@@ -0,0 +1,386 @@
...
+            .map(signerIdentifier -> "0x" + signerIdentifier.toStringIdentifier())
+            .findAny()
+            .orElseThrow();
+  }
+
+  Web3j jsonRpc() {
+    return jsonRpc;
+  }
+
+  Eea eeaJsonRpc() {
```
Comment (siladu) [inline core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java:174]: Not used

File: core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java [hunk 7]
```diff
@@ -0,0 +1,386 @@
...
+    clientAndServer
+        .when(request().withBody(json(request.getBody())), exactly(1))
+        .respond(
+            response()
+                .withBody(response.getBody())
+                .withHeaders(MockServer.headers(response.getHeaders()))
+                .withStatusCode(response.getStatusCode()));
+  }
+
+  void setupEthNodeResponse(
```
Comment (siladu) [inline core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java:205]: not used

File: core/src/main/java/tech/pegasys/web3signer/core/Eth1AddressSignerProvider.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+/** Wrapper on SignerProvider that uses address (Eth1AddressSignerIdentifier) to load signer */
+public class Eth1AddressSignerProvider {
+  private final SignerProvider signerProvider;
+
+  public Eth1AddressSignerProvider(final SignerProvider signerProvider) {
+    this.signerProvider = signerProvider;
+  }
+
+  /* Gets a signer from its address, address is expected to be hex value, with or without 0x */
+  public Optional<Signer> getSigner(final String address) {
```
Comment (siladu) [inline core/src/main/java/tech/pegasys/web3signer/core/Eth1AddressSignerProvider.java:31]: not used

File: core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/RpcUtil.java [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+import tech.pegasys.web3signer.core.service.jsonrpc.response.JsonRpcErrorResponse;
+
+import java.math.BigInteger;
+import java.util.List;
+
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.DecodeException;
+import io.vertx.core.json.JsonObject;
+
+public class RpcUtil {
```
Comment (siladu) [inline core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/RpcUtil.java:27]: Not used?

File: core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/handlers/HttpResponseFactory.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+  public void response(
+      final HttpServerResponse response, final int statusCode, final JsonRpcResponse body) {
+    response.putHeader("Content", JSON);
+    response.setStatusCode(statusCode);
+    response.setChunked(false);
+    response.end(Json.encodeToBuffer(body));
+  }
+
+  public void successResponse(
```
Comment (siladu) [inline core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/handlers/HttpResponseFactory.java:37]: not used

File: core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/response/JsonRpcError.java
Comment (siladu) [inline core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/response/JsonRpcError.java:1]: Is the plan to tidy this up at the end? Maybe privacy stuff could go already?

File: core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java (hunk 2 above)
Comment (jframe) [reply core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java:137]: not needed anymore. removed it

File: core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java (hunk 3 above)
Comment (jframe) [reply core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java:123]: removed it

File: core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java (hunk 5 above)
Comment (jframe) [reply core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java:162]: removed

File: core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java (hunk 4 above)
Comment (jframe) [reply core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java:131]: removed

File: core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java (hunk 3 above)
Comment (jframe) [reply core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java:123]: removed

File: core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java (hunk 7 above)
Comment (jframe) [reply core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java:205]: removed

File: core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/DefaultTestBase.java (hunk 1 above)
Comment (siladu) [reply core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/DefaultTestBase.java]: If creating the file as new in web3signer would set it as this year then maybe, but probably not an issue either way

**Review by siladu — APPROVED** [review]
LGTM

**Pull request merged**